src/core/blockchain/zilliqa/client-utils.ts

```diff
@@ -0,0 +1,126 @@
+import { IClientUtils } from '../types/client-utils';
+import { Client } from './client';
+import { IBlockchainTransaction, Blockchain, TransactionType } from '../types';
+import { TokenType, TokenScreenComponentType } from '../types/token';
+import { getAddressFromPublicKey, toBech32Address } from '@zilliqa-js/crypto';
+import { isBech32 } from '@zilliqa-js/util/dist/validation';
+import { config } from './config';
+import { ITokenConfigState } from '../../../redux/tokens/state';
+import { TransactionStatus } from '../../wallet/types';
+import { generateAccountTokenState } from '../../../redux/tokens/static-selectors';
+
+export class ClientUtils implements IClientUtils {
+    constructor(private client: Client) {}
+
+    async getTransaction(hash: string): Promise<IBlockchainTransaction> {
+        const txData = await this.client.http.jsonRpc('GetTransaction', [hash]).then(response => {
+            if (!response.result) {
+                throw new Error(
```

Comment:
when returning a promise it's better to return a `Promise.reject()` than throwing an exception.